client/asset/bch/spv.go

```diff
@@ -0,0 +1,1055 @@
+	err := os.MkdirAll(logDir, 0744)
+	if err != nil {
+		return fmt.Errorf("error creating wallet directories: %w", err)
+	}
+
+	loader := wallet.NewLoader(net, netDir, true, 250)
+
+	pubPass := []byte(wallet.InsecurePubPassphrase)
+
+	btcw, err := loader.CreateNewWallet(pubPass, privPass, seed, bday)
+	if err != nil {
+		return fmt.Errorf("CreateNewWallet error: %w", err)
+	}
+
+	errCloser := dex.NewErrorCloser(log)
+	defer errCloser.Done()
+	errCloser.Add(loader.UnloadWallet)
+
+	if extIdx > 0 || intIdx > 0 {
+		err = extendAddresses(extIdx, intIdx, btcw)
+		if err != nil {
+			return fmt.Errorf("failed to set starting address indexes: %w", err)
+		}
+	}
+
+	// The chain service DB
+	neutrinoDBPath := filepath.Join(netDir, neutrinoDBName)
+	db, err := walletdb.Create("bdb", neutrinoDBPath, true)
+	if err != nil {
+		return fmt.Errorf("unable to create neutrino db at %q: %w", neutrinoDBPath, err)
+	}
+	if err = db.Close(); err != nil {
+		return fmt.Errorf("error closing newly created wallet database: %w", err)
+	}
+
+	if err := loader.UnloadWallet(); err != nil {
+		return fmt.Errorf("error unloading wallet: %w", err)
+	}
+
+	errCloser.Success()
+	return nil
+}
+
+// Start initializes the *bchwallet.Wallet and its supporting players and starts
+// syncing.
+func (w *bchSPVWallet) Start() (btc.SPVService, error) {
+	if err := logNeutrino(w.dir, w.log); err != nil {
+		return nil, fmt.Errorf("error initializing btcwallet+neutrino logging: %v", err)
+	}
+	// recoverWindow arguments borrowed from bchwallet directly.
+	w.loader = wallet.NewLoader(w.chainParams, w.dir, true, 250)
+
+	exists, err := w.loader.WalletExists()
+	if err != nil {
+		return nil, fmt.Errorf("error verifying wallet existence: %v", err)
+	}
+	if !exists {
+		return nil, errors.New("wallet not found")
+	}
+
+	w.log.Debug("Starting native BCH wallet...")
+	w.Wallet, err = w.loader.OpenExistingWallet([]byte(wallet.InsecurePubPassphrase), false)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't load wallet: %w", err)
+	}
+
+	errCloser := dex.NewErrorCloser(w.log)
+	defer errCloser.Done()
+	errCloser.Add(w.loader.UnloadWallet)
+
+	neutrinoDBPath := filepath.Join(w.dir, neutrinoDBName)
+	w.neutrinoDB, err = walletdb.Create("bdb", neutrinoDBPath, true)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create wallet db at %q: %v", neutrinoDBPath, err)
+	}
+	errCloser.Add(w.neutrinoDB.Close)
+
+	// Depending on the network, we add some addpeers or a connect peer. On
+	// regtest, if the peers haven't been explicitly set, add the simnet harness
+	// alpha node as an additional peer so we don't have to type it in. On
+	// mainet and testnet3, add a known reliable persistent peer to be used in
+	// addition to normal DNS seed-based peer discovery.
+	// var addPeers []string
+	var connectPeers []string
+	switch w.chainParams.Net {
+	case bchwire.TestNet, bchwire.SimNet: // plain "wire.TestNet" is regnet!
+		connectPeers = []string{"localhost:21577"}
+	}
```

Comment:
I can't seem to get past "no peers" on testnet. I have a local node running and tried changing this to the port I set for it but still no peers. Any idea what I can try?

Reply:
The comment here I think is to indicate that there is some ambiguity between testnet 1 and simnet. As you indicated, we're using `bchwire.TestNet4`. I've now added a `addPeers = []string{"localhost:28333"}` for that case, and was able to sync with `bchd --testnet4` running locally. For Bitcoin, we added a testnet btcd node at `dex-test.ssgen.io`. Maybe we'll want to do the same for bch eventually.

Reply:
Might be needed if cfilter peers are scarce.  Hopefully not though

Reply:
Oh no, I was using -testnet instead of -testnet4. So... I should use testnet 4 right?

Reply:
Oh crap. We can't use testnet 4, can we?

Reply:
Pretty sure we sorted it out on Matrix, but we're on testnet4 so that's right.  @JoeGruffins I guess the confusion for you was that you need bchd installed locally to have a node serving cfilters since bitcoin-cash-node isn't able.

Reply:
Still not finding peers after running bchd. This definitely works for others on testnet?

Reply:
> Still not finding peers after running bchd. This definitely works for others on testnet?

Only when running local bchd.  Like:

```sh
bchd --testnet4 --whitelist 127.0.0.0/8 --whitelist ::1 --banduration=20s
```